internal/cmds/cmds.go

```diff
@@ -422,16 +434,89 @@ func copyFiles(ctx log.Logger, fileExtensionSuffix string, extensionSubdirectory
 				errMessage := fmt.Sprintf("Failed to copy '%s' file '%s' to path '%s'. Contact ICM team AzureRT\\Extensions for this service error.",
 					fileExtensionSuffix, sourceFileFullPath, destinationFileFullPath)
 				ctx.Log("message", errMessage)
-				return errors.Wrapf(copyError, errMessage)
+				return fileNamesMigrated, errors.Wrapf(copyError, errMessage)
 			} else {
 				ctx.Log("message", fmt.Sprintf("File '%s' was copied successfully to '%s'", sourceFileFullPath, destinationFileFullPath))
 				numberOfFilesMigrated++
+				fileNamesMigrated.PushBack(fileName)
 			}
 		}
 	}
 
 	ctx.Log("message", fmt.Sprintf("Migrated %d '%s' files from extension version '%s' to '%s'", numberOfFilesMigrated, fileExtensionSuffix, oldExtensionVersion, newExtensionVersion))
 
+	return fileNamesMigrated, nil
+}
+
+// This need to be only executed by Update operation
+func createDummyStatusFilesIfNeeded(ctx log.Logger, mrseqFilesNameList *list.List) error {
+	if mrseqFilesNameList == nil || mrseqFilesNameList.Len() <= 0 {
+		return nil
+	}
+
+	// Create dummy status file for .mrseq file if status file is not available.
+	newExtensionDirectory := os.Getenv(constants.ExtensionPathEnvName)
+	statusFileDirectoryPath := filepath.Join(newExtensionDirectory, constants.StatusFileDirectory)
+
+	var mrSeqFileName string
+	var mrSeqFileFullPath string
+	var extensionName string
+	var mrSeqFileExtensionIndex int
+	var statusFileName string
+	var statusFilePath string
+	for mreSeqFileNameElement := mrseqFilesNameList.Front(); mreSeqFileNameElement != nil; mreSeqFileNameElement = mreSeqFileNameElement.Next() {
+		mrSeqFileName = (mreSeqFileNameElement.Value).(string)
```

Comment:
should this be best effort? If we fail to read one .mrseq, should we just move past that one and try to write dummy files for the others?

Reply:
Yes, agree